Test/DependencyCollector/FunctionalTests/FuncTest/SqlTests.cs

```diff
@@ -372,7 +387,7 @@ private void ExecuteSyncSqlTests(TestWebApplication testWebApplication, int expe
                     //// The above request would have trigged RDD module to monitor and create RDD telemetry
                     //// Listen in the fake endpoint and see if the RDDTelemtry is captured
 
-                    var allItems = DeploymentAndValidationTools.SdkEventListener.ReceiveAllItemsDuringTimeOfType<TelemetryItem<RemoteDependencyData>>(DeploymentAndValidationTools.SleepTimeForSdkToSendEvents);
+                    var allItems = DeploymentAndValidationTools.SdkEventListener.ReceiveAllItemsDuringTimeOfType<TelemetryItem<RemoteDependencyData>>(DeploymentAndValidationTools.SleepTimeForSdkToSendEvents).ToArray();
```

Comment:
It is easier to debug this way and we do not need to be performant in functional tests

---
In reply to: [82121011](https://github.com/Microsoft/ApplicationInsights-dotnet-server/pull/170#discussion_r82121011) [](ancestors = 82121011)